### PR TITLE
Improve README section for running tests

### PR DESCRIPTION
**Pull request by Martchus**
* Add more details about `tox` environments
* Mention how to ensure correct formatting of files
* Keep information related to direct `pytest` invocations together (and
  not mention how to generate the coverage report in the middle)
* Use subsections for the different parts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,16 @@ all environments:
 tox
 ```
 
-Or you can run the tests via pytest with one of the following commands:
+It is also possible to run only a special environment, e.g. `tox -e flake8` for
+the linter, `tox -e py38` for running tests at the specified Python version (the
+interpreter needs to be installed at this version) or `tox -e cov` for the
+coverage report (see details below).
+
+To ensure files are passing the style check, invoke `black .` to apply correct
```
Comment (perlpunk) [inline README.md:73]: That should be `tox -e black` for consistency.
The actual commands and options can be found in `tox.ini` also, if one wants to call them directly.

Comment (Martchus) [reply README.md:73]: Yes, but when trying to invoke it like this I'm running into https://progress.opensuse.org/issues/102584. Maybe I can figure this issue out, though.

Comment (Martchus) [reply README.md]: Fixed. I also added a comment how I've got rid of the error in the README because it seems it isn't just me who experienced it.

**Comment by Martchus** [comment]
@perlpunk I've changed the black command back because the `tox -e black` actually only checks whether the source files are correctly formatted. However, here I want to show the command for actually formatting them.

File: README.md [hunk 2]
```diff
@@ -65,7 +65,21 @@ all environments:
 tox
 ```
 
-Or you can run the tests via pytest with one of the following commands:
+It is also possible to run only a special environment, e.g. `tox -e flake8` for
+the linter, `tox -e py38` for running tests at the specified Python version (the
+interpreter needs to be installed at this version) or `tox -e cov` for the
```
Comment (okurz) [inline README.md:70]: s/at this version/for this version/?

**Review by okurz — COMMENTED** [review]
one trivial question, rest is fine, +1 in general

**Review by okurz — APPROVED** [review]

**Pull request merged**